dwave/optimization/include/dwave-optimization/graph.hpp

```diff
@@ -29,13 +29,15 @@
 #include "dwave-optimization/array.hpp"
 #include "dwave-optimization/state.hpp"
 #include "dwave-optimization/utils.hpp"
+#include "xxhash.h"
```

Comment:
Because we distribute this library and because it can (in theory) be built as a standalone C++ library, I am not stoked about having a third party library outside the compilation barrier. IMO we should move all of the `xxhash` stuff into the `.cpp` file. You can still store the table on the class via an opaque pointer.

In general I am not stoked about introducing a third-party dependency just for this one feature. But still ruminating on that.